### PR TITLE
Client headers refactoring

### PR DESCRIPTION
**Pull request by LavrovArtem**

### CODE REVIEW

File: src/request-pipeline/builtin-header-names.ts [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+    accessControlAllowOrigin:         'access-control-allow-origin',
+    accessControlAllowCredentials:    'access-control-allow-credentials',
+    contentSecurityPolicy:            'content-security-policy',
+    contentSecurityPolicyReportOnly:  'content-security-policy-report-only',
+    xContentSecurityPolicy:           'x-content-security-policy',
+    xContentSecurityPolicyReportOnly: 'x-content-security-policy-report-only',
+    xWebkitCsp:                       'x-webkit-csp'
+};
+
+// xhr/headers
```
Comment (LavrovArtem) [inline src/request-pipeline/builtin-header-names.ts:44]: remove

File: src/client/utils/headers.ts [hunk 2]
```diff
@@ -0,0 +1,62 @@
+import BUILTIN_HEADERS from '../../request-pipeline/builtin-header-names';
+import INTERNAL_HEADERS from '../../request-pipeline/internal-header-names';
+
+export function transformRequestHeaderNameToInternal (headerName: any) {
+    if (typeof headerName === 'string') {
+        headerName = headerName.toLowerCase();
+
+        if (headerName === BUILTIN_HEADERS.authorization)
```
Comment (miherlosev) [inline src/client/utils/headers.ts:8]: You can create predefined dictionaries for transformations:
```js
const REQUEST_HEADER_TO_INTERNAL = {
   [BUILTIN_HEADERS.authorization]: BUILTIN_HEADERS.authorization,
   ...
}

const RESPONSE_HEADER_TO_INTERNAL = {...}

export function transformRequestHeaderNameToInternal (headerName: any) {
   return REQUEST_HEADER_TO_INTERNAL[headerName] || headerName;
}

```

**Comment by LavrovArtem** [comment]
@testcafe-build-bot retest

**Comment by LavrovArtem** [comment]
@testcafe-build-bot retest

**Comment by LavrovArtem** [comment]
@testcafe-build-bot retest

**Comment by LavrovArtem** [comment]
@testcafe-build-bot retest

**Comment by LavrovArtem** [comment]
@testcafe-build-bot retest

**Comment by miherlosev** [comment]
@testcafe-build-bot retest

**Comment by miherlosev** [comment]
@testcafe-build-bot retest

**Comment by miherlosev** [comment]
@testcafe-build-bot retest

**Comment by miherlosev** [comment]
@testcafe-build-bot retest

**Comment by miherlosev** [comment]
@testcafe-build-bot retest

**Pull request merged**